eng/common/TestResources/Remove-TestResources.ps1

```diff
@@ -126,18 +131,27 @@ if ($ProvisionerApplicationId) {
 $context = Get-AzContext
 
 if (!$ResourceGroupName) {
-    # Make sure $BaseName is set.
-    if (!$BaseName) {
-        $UserName = if ($env:USER) { $env:USER } else { "${env:USERNAME}" }
-        # Remove spaces, etc. that may be in $UserName
-        $UserName = $UserName -replace '\W'
-
-        $BaseName = "$UserName$ServiceDirectory"
-        Log "BaseName was not set. Using default base name '$BaseName'"
-    }
+    if ($CI) {
+        $envVarName = (BuildServiceDirectoryPrefix (GetServiceName $ServiceDirectory)) + "RESOURCE_GROUP"
+        $ResourceGroupName = [Environment]::GetEnvironmentVariable($envVarName)
+        if (!$ResourceGroupName) {
+            Write-Error "Could not find resource group name environment variable '$envVarName'"
+            exit 1
+        }
+    } else {
+        # Make sure $BaseName is set.
+        if (!$BaseName) {
+            $UserName = if ($env:USER) { $env:USER } else { "${env:USERNAME}" }
+            # Remove spaces, etc. that may be in $UserName
+            $UserName = $UserName -replace '\W'
+
+            $BaseName = "$UserName$ServiceDirectory"
+            Log "BaseName was not set. Using default base name '$BaseName'"
```

Comment:
Nit: Given we do this in New-TestResoruces.ps1, I wonder if this should be moved to the helpers.

Reply:
Good idea.